data/scripts.js

```diff
@@ -632,32 +722,80 @@ let BattleScripts = {
 				this.add('-fail', pokemon);
 				this.attrLastMove('[still]');
 			}
-			return false;
+			return [[false], targets]; // single-target only
```

Comment:
https://www.smogon.com/forums/posts/8110059

Synchronoise used to use `TryHit` single event to mark the move as immune if the target did not share a type with the user. However this event now only runs for the first Pokémon, which means that it hits no Pokémon at all depending on whether the ally does not share a type, unless it misses the ally due to semi-invulnerability, which is in itself a bug which I forgot to file, as synchronise should never miss an immune target.

Checking other moves, it looks like Captivate is also expecting a `TryHit` single event for each target. I don't know what the on-cart mechanics of Captivate are when only one foe has a suitable gender, or what happens if the unsuitable foe is semi-invulnerable.

Speaking of semi-invulnerability, when I tried to reproduce this, I happened to use Toxic from a Poison type to a semi-invulnerable target. It missed. This is also incorrect behaviour; since Gen 6, this should work like No Guard.